examples/data-objects/dataStoreRace/src/index.tsx

```diff
@@ -0,0 +1,248 @@
+        const facts: IChannelFactory[] = [
+            SharedMap.getFactory(),
+            SharedCounter.getFactory(),
+        ];
+        const reg: ISharedObjectRegistry =
+            new Map<string, IChannelFactory | undefined>(facts.map((ext) => [ext.type, ext]));
+
+        const dsRuntime = FluidDataStoreRuntime.load(
+            context,
+            reg);
+
+        // cannot await this as we get in infinite loop of realization due to handle resolution
+        //
+        const clickerP = MergeableClicker.load(dsRuntime);
+
+        dsRuntime.registerRequestHandler(async (req) => (await clickerP).request(req));
+
+        return dsRuntime;
+    },
+    createLocal: async (containerRuntime: ContainerRuntime) => {
+        const loader = Loader._create(
+            [],
+            [],
+            { load: async () => ({ fluidExport: NormalContainerFactory }) },
+            {},
+            { },
+            new Map(),
+        );
+        const container = await loader.createDetachedContainer(containerRuntime.codeDetails);
+        const res = await container.request({ url: "" });
+        return res.value as MergeableClicker;
+    },
+};
+
+export const NormalContainerFactory: IRuntimeFactory =  {
+    get IRuntimeFactory() {return this;},
+    instantiateRuntime: async (context: IContainerContext)=>{
+        const runtime: ContainerRuntime =  await ContainerRuntime.load(
+            context,
+            [["default", Promise.resolve(MerableClickerInstantiationFactory)]],
+            async (req)=>{
+                const rp = RequestParser.create(req);
+                const ds =  await runtime.getRootDataStore(rp.pathParts[0] ?? "default", true);
+                return ds.request(rp.pathParts.length > 1 ? rp.createSubRequest(1) : { url: "" });
+            },
+            {
+                enableSummarizerNode: false,
+                generateSummaries: false,
+            });
+
+        if (runtime.existing === false) {
+            await runtime.createRootDataStore("default", "default");
+        }
+
+        return runtime;
+    },
+};
+
+export const RaceContainerFactory: IRuntimeFactory = {
+    get IRuntimeFactory() {return this;},
+    instantiateRuntime: async (context: IContainerContext)=>{
+        let current: MergeableClicker;
+        let lastElm: HTMLElement;
+        const viewProxy: IFluidHTMLView = {
+            get IFluidHTMLView() {return this;},
+            render(elm: HTMLElement, options?: IFluidHTMLOptions) {
+                lastElm = elm;
+                current.render(elm);
+            },
+        };
+
+        const runtime =  await ContainerRuntime.load(
+            context,
+            [["default", Promise.resolve(MerableClickerInstantiationFactory)]],
+            async (req)=> {
+                return  { status: 200, mimeType: "fluid/object", value: viewProxy };
+            });
+
+        const local = current = await MerableClickerInstantiationFactory.createLocal(runtime);
```

Comment:
this is the meat of the prototype and shows how a client could use this to race. We should check if it exists before we race, but for the prototype i want to exercise race on every load